library/psa_crypto_core.h

```diff
@@ -55,6 +55,20 @@ typedef enum {
     PSA_SLOT_PENDING_DELETION,
 } psa_key_slot_state_t;
 
+/* If static key slots are enabled, determine the buffer size of each key slot.
+ * This size depends on the enabled PSA key types. If the key type does not
+ * have a specific PSA_WANT to determine the maximum key length (ex: AES,
+ * Camellia, ARIA) then pick the maximum supported key length for that key
+ * type. */
+#if defined(MBEDTLS_PSA_STATIC_KEY_SLOTS)
+
+#define MAX3(a, b, c) ((a) < (b) ? (b) < (c) ? (c) : (b) : (a) < (c) ? (c) : (a))
+
+#define KEY_SLOT_BUFFER_SIZE        \
+        MAX3(PSA_EXPORT_KEY_PAIR_OR_PUBLIC_MAX_SIZE, PSA_HASH_MAX_SIZE, PSA_CIPHER_MAX_KEY_LENGTH)
```

Comment:
This does not account for all key types. In practice, on devices that want to avoid dynamic allocation, it's likely that public keys will be the largest thing unless asymmetric cryptography is completely disabled, and hashes otherwise. But that's still an arbitrary limitation that would need to be at the very least documented, and perhaps we should give a way to configure it.

Reply:
> This does not account for all key types

Yeah, I was aware of this limitation. For the list of supported key types I get it from `crypto_values.h` and I think I've included support for `HMAC, AES, ARIA, DES, CAMELLIA, CHACHA20, RSA_PUBLIC_KEY, RSA_KEY_PAIR, ECC_PUBLIC_KEY_BASE, ECC_KEY_PAIR_BASE, DH_PUBLIC_KEY_BASE, DH_KEY_PAIR_BASE`.
I skipped RAW_DATA` because IMO it didn't make much sense for what I was trying to achieve, while for `DERIVE, PASSWORD, PASSWORD_HASH, PEPPER` I didn't know exactly what to set. 
Do you think it's worth to add support also for the missing key types? Or perhaps we can just move to the 2nd point (because it has a possible alternative to this)

> and perhaps we should give a way to configure it.

Do you mean that I should add another build symbol to set the max slot size manually? In this case it can have a default value (for example `PSA_EXPORT_KEY_PAIR_OR_PUBLIC_MAX_SIZE`), but then the user can change it at build time if he knows that he's going to use a larger key for some reason. This would also greatly simplify the previous part of this comment and allow to include also `RAW_DATA` key type. Is my understanding correct?

Reply:
There's no intrinsic upper limit for things like `RAW_DATA` and `DERIVE`. In practice, `DERIVE` is usually 128 or 256 bits for root secrets and intermediate keys, and the size of the largest elliptic curves for secrets from key agreement. `RAW_DATA` could be anything, for example it's often used to store a certificate in a secure enclave alongside the corresponding private key. So yes, for practicality, a numerical setting that defaults to `PSA_EXPORT_PUBLIC_KEY_MAX_SIZE` is probably the right thing.

Reply:
Another case is wrapped keys stored on behalf of a secure element. These have the size of the key plus some overhead, typically around 32 to 64 bytes.